framework/src/modules/interoperability/utils.ts

```diff
@@ -593,3 +618,347 @@ export const commonCCUExecutelogic = async (args: CommonExecutionLogicArgs) => {
+
+			const { blsKey } = activeValidator;
+			if (j < activeValidators.length - 1 && blsKey.compare(activeValidators[j + 1].blsKey) >= 0) {
+				throw new Error(
+					'Active validators must be ordered lexicographically by blsKey property and pairwise distinct.',
+				);
+			}
+
+			const { bftWeight } = activeValidator;
+			totalWeight += bftWeight;
+		}
+
+		if (totalWeight > MAX_UINT64) {
+			throw new Error(
+				'The total BFT weight of all active validators has to be less than or equal to MAX_UINT64.',
+			);
+		}
+
+		const checkBftWeightValue = totalWeight / BigInt(3) + BigInt(1);
+		if (checkBftWeightValue > totalWeight || checkBftWeightValue > certificateThreshold) {
+			throw new Error('The total BFT weight of all active validators is not valid.');
+		}
+
+		await chainValidatorsStore.setWithSchema(
+			chainValidators.storeKey,
+			chainValidators.storeValue,
+			chainValidatorsSchema,
+		);
+	}
+
+	const chainDataStoreKeySet = new dataStructures.BufferSet();
+	const chainDataStore = getStore(id, STORE_PREFIX_CHAIN_DATA);
+	let isAnotherSidechainAccount = 0;
+	for (const chainData of genesisStore.chainDataSubstore) {
+		const chainDataStoreKey = chainData.storeKey;
+		if (chainDataStoreKeySet.has(chainDataStoreKey)) {
+			throw new Error(`Chain data store key ${chainDataStoreKey.toString('hex')} is duplicated.`);
+		}
+		chainDataStoreKeySet.add(chainDataStoreKey);
+
+		const chainAccountStatus = chainData.storeValue.status;
+		if (chainAccountStatus === CHAIN_TERMINATED) {
+			if (outboxRootStoreKeySet.has(chainDataStoreKey)) {
+				throw new Error('Outbox root store cannot have entry for a terminated chain account.');
+			}
+			if (
+				!channelDataStoreKeySet.has(chainDataStoreKey) ||
+				!chainValidatorsStoreKeySet.has(chainDataStoreKey)
+			) {
+				throw new Error(
+					`Chain data store key ${chainDataStoreKey.toString(
+						'hex',
+					)} missing in some or all of channel data and chain validators stores.`,
+				);
+			}
+		}
+		if (
+			!outboxRootStoreKeySet.has(chainDataStoreKey) ||
+			!channelDataStoreKeySet.has(chainDataStoreKey) ||
+			!chainValidatorsStoreKeySet.has(chainDataStoreKey)
+		) {
+			throw new Error(
+				`Chain data store key ${chainDataStoreKey.toString(
+					'hex',
+				)} missing in some or all of outbox root, channel data and chain validators stores.`,
+			);
+		}
+
+		if (
+			!(
+				chainDataStoreKey.equals(ownChainAccount.id) ||
+				chainDataStoreKey.equals(MAINCHAIN_ID_BUFFER)
+			)
+		) {
+			isAnotherSidechainAccount = 1;
+		}
+
+		await chainDataStore.setWithSchema(
+			chainData.storeKey,
+			chainData.storeValue,
+			chainAccountSchema,
+		);
+	}
+
+	if (
+		isAnotherSidechainAccount &&
+		!(chainDataStoreKeySet.has(MAINCHAIN_ID_BUFFER) && chainDataStoreKeySet.has(ownChainAccount.id))
+	) {
+		throw new Error(
+			'If a chain account for another sidechain is present, then a chain account for the mainchain must be present, as well as the own chain account.',
+		);
+	}
```

Comment:
If a chain account for another sidechain is present, than a chain account for the mainchain (chainID = 1) must be present, as well as the own chain account